scripts/e2e.js

```diff
@@ -0,0 +1,187 @@
+
+    const TestToken = await ethers.getContractFactory('TestToken');
+    console.log('Deploying TestToken contract...');
+    const token = await TestToken.deploy("1000000");
+    await token.deployed();
+    console.log('TestToken deployed to:', token.address);
+
+    console.log(await peg.activateDeposits());
+
+    /***************** Make Deposit on ETHEREUM **********************/
+
+    let depositAmount = 5644;
+    // Alice
+    let cennznetAddress = "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d";
+    console.log(await token.approve(peg.address, depositAmount));
+    const deposition = await peg.deposit(token.address, depositAmount, cennznetAddress);
+    const depositTxHash = deposition.hash;
+    console.log('depositTxHash:::',depositTxHash);
+
+
+
+    /***************** CENNZnet Setup **********************/
+        // connecting to local chain
+    const api = await Api.create();
+    const keyring = new Keyring({type: 'sr25519'});
+    const alice = keyring.addFromUri('//Alice');
+    const transaction1 = api.tx.erc20Peg.activateDeposits(true);
+    const transaction2 = api.tx.erc20Peg.activateWithdrawals(true);
+    const transaction3 = api.tx.erc20Peg.setContractAddress(peg.address);
+    const transaction4 = api.tx.ethBridge.setEventConfirmations(0); // Hardhat only makes blocks when txs are sent
+    const batchBridgeActivationEx = api.tx.utility.batch([
+        transaction1,
+        transaction2,
+        transaction3,
+        transaction4
+    ]);
+
+    await new Promise(async (resolve, reject) => {
+        await api.tx.sudo.sudo(batchBridgeActivationEx).signAndSend(alice, async ({status, events}) => {
+            if (status.isInBlock) {
+                events.forEach(({phase, event: {data, method, section}}) => {
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                    resolve();
+                });
+            }
+        });
+    });
+
+    const testTokenId = await api.query.genericAsset.nextAssetId();
+    const claim = {
+        tokenAddress: token.address,
+        amount: depositAmount,
+        beneficiary: cennznetAddress
+    };
+
+    console.log('testTokenId::', testTokenId.toString());
+
+    /***************** Make Deposit on CENNZnet **********************/
+
+    let nonce = await api.rpc.system.accountNextIndex(alice.address);
+    await new Promise(async (resolve, reject) => {
+        await api.tx.erc20Peg.depositClaim(depositTxHash, claim).signAndSend(alice, {nonce}, async ({status, events}) => {
+            if (status.isInBlock) {
+                for (const {event: {method, section, data}} of events) {
+                    console.log('\t', `: ${section}.${method}`, data.toString());
+                    if (section === 'erc20Peg' && method == 'Erc20Claim') {
+                        console.log('*******************************************');
```

Comment:
this only means the claim started.
need to check future block events for `Erc20ClaimSuccess` with the matching Id
also best to match on the `sender` from the event is the same also